### PR TITLE
fix(es/compat): transform `logical and assignment` (`&&=`) operator

### LINKED ISSUE #3224: Does not transform `&&=` “logical AND assignment” operator
### Describe the bug

(similar to #1788) swc does not transform the [Logical AND assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_AND_assignment) operator (`&&=`).

### Input code

```typescript
const swc = require('@swc/core');
swc.transformSync(`a ||= b`, {});  // a || (a = b);
swc.transformSync(`a &&= b`, {});  // a &&= b;
```

### Config
N/A

### Expected behavior

The `&&=` operator is transformed by SWC, like the `||=` and `??=` operators are.

### Actual behavior

The `&&=` operator is left untransformed.

### Version

1.2.127

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by magic-akari**
**Description:**

**BREAKING CHANGE:**

**Related issue (if exists):**
- fix #3224

### CODE REVIEW

**Review by kdy1 — DISMISSED** [review]
LGTM. Thank you!

File: crates/swc_ecma_transforms_compat/src/es2021/logical_assignments.rs [hunk 1]
```diff
@@ -33,6 +33,12 @@ impl VisitMut for Operators {
 
         match e {
             Expr::Assign(AssignExpr {
+                span,
+                op: op @ op!("&&="),
```
Comment (Austaras) [inline crates/swc_ecma_transforms_compat/src/es2021/logical_assignments.rs:37]: `op @op!("&&=) | op @op!("||=") | op @op!("??=")` would be better

Comment (kdy1) [reply crates/swc_ecma_transforms_compat/src/es2021/logical_assignments.rs:37]: Oh you are right. Thank you!

**Review by kdy1 — APPROVED** [review]
LGTM. Thanks!

**Comment by tranvansang** [comment]
Isn't the doc not up-to-date?

Currently, I haven't migrated from babel to swc due to this?

https://swc.rs/docs/migrating-from-babel

image

**Comment by kdy1** [comment]
Yeah the doc is outdated

**Comment by kdy1** [comment]
It's not stage 1

**Comment by tranvansang** [comment]
> It's not stage 1

Is ✘ a correct status?

This is a very popular feature, if not being supported it should be prioritized.

**Comment by kdy1** [comment]
No, it's already supported because it's not stage 1 anymore

**Pull request merged**